### PR TITLE
packaging: `pyproject.toml` + `setuptools_scm`

### PR DESCRIPTION
**Pull request by yunjunz**
This PR updates the packaging, following https://github.com/insarlab/PySolid/pull/43.

+ replace `setup.py` with `pyproject.toml` file

+ replace `version.py` with `setuptools_scm` within `pyproject.toml` file

+ requirements.txt: add packaging-related dependencies

### CODE REVIEW

**Comment by yunjunz** [comment]
I plan to cut a release after this PR is merged, because of the `numpy` bugfix in #17.

File: src/pyaps3/__init__.py [hunk 1]
```diff
@@ -7,13 +7,21 @@
     Details of the python module can be obtained `here. <http://code.google.com/p/pyaps>`_
 '''
 
-__all__ = ['autoget','objects']
+# get version info
+from importlib.metadata import PackageNotFoundError, version
```
Comment (jhkennedy) [inline src/pyaps3/__init__.py:11]: `importlib.metadata` is only available for Python >=3.8. For python <3.8 you'll need the `importlib_metadata` package, which backports the functionality, and do:
```python
from importlib_metadata import PackageNotFoundError, version
```

I would just drop support for Python <3.8 and set the `requires-python = ">=3.8"` here:
https://github.com/insarlab/PyAPS/pull/19/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R13
```

File: src/pyaps3/__init__.py [hunk 2]
```diff
@@ -7,13 +7,21 @@
...
+          'Or, to just get the version number, use:\n'
+          '   python -m setuptools_scm')
 
+# top-level functions
 from pyaps3.objects import PyAPS
 from pyaps3.autoget import ECMWFdload, MERRAdload, NARRdload
 
-# get version info
-from pyaps3.version import release_version as __version__
+__all__ = ['autoget', 'objects']
```
Comment (jhkennedy) [inline src/pyaps3/__init__.py:24]: I think you'll either need:
```python
from pyaps3 import objects
from pyaps3 import autoget

__all__ = ['autoget', 'objects']
```
or:

```python
from pyaps3.objects import PyAPS
from pyaps3.autoget import ECMWFdload, MERRAdload, NARRdload

__all__ = ['PyAPS', 'ECMWFdload', 'MERRAdload', 'NARRdload']
```

`__all__` defines what'll be available with a `from pyaps3 import *` and will affect how IDE's interpret the package API. 

You may also want to include `__version__` in `__all__`.

**Review by jhkennedy — CHANGES_REQUESTED** [review]
Overall looks good! Couple of minor changes to consider

File: src/pyaps3/__init__.py (hunk 1 above)
Comment (yunjunz) [reply src/pyaps3/__init__.py:13]: Thank you @jhkennedy. Dropped python<3.8 as suggested.

File: src/pyaps3/__init__.py (hunk 2 above)
Comment (yunjunz) [reply src/pyaps3/__init__.py]: I totally missed this. Thank you! Fixed it now.

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**